### ISSUE TITLE
Mail attachments with the same name without extension = infinite loop

### ISSUE THREAD

**Issue by BorisNovg** [glpi-project/glpi#9667]
**Describe the bug**

A message with attachments with a file name that does not have an extension leads to an infinite loop when sent to the mail receiver.

**To reproduce**

1. Configure the mail receiver.
2. Prepare a file named 1234567890 (without extension!) and any content.
3. Create a new email in Microsoft Outlook (in HTML format), attach the file 1234567890 to the message, attach the same file 1234567890 twice more to the message, send this message (with three identical attachments with the file name without extension) to the address of your mail receiver from point 1.
4. Go to the Home-Setup-Receivers-(yourMailReceiver)-Actions page and click "Get email tickets now".
5. See an infinite loop. The ticket has not been created. The test message has not been deleted from the mailbox.

**Expected behavior**

A new ticket has been created with the text and attachments of the email you sent.

**Logs**

```
[2021-10-06 11:23:29] glpiphplog.CRITICAL: *** PHP Error (1): Maximum execution time of 600 seconds exceeded in /usr/share/glpi/inc/mailcollector.class.php at line 1592
```

**Your GLPI setup (you can find it in Setup > General menu, System tab)**

```
[code]
 
GLPI 9.5.6 ( => /usr/share/glpi)
Installation mode: RPM

Server
 
Operating system: Linux help.admreg.nov.ru 3.10.0-1160.42.2.el7.x86_64 #1 SMP Tue Sep 7 14:49:57 UTC 2021 x86_64
PHP 8.0.11 apache2handler (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apcu, bcmath, bz2,
	calendar, ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, hash, iconv, imap, intl, json, ldap, libxml,
	mbstring, mysqli, mysqlnd, openssl, pcre, pdo_mysql, pdo_sqlite, posix, session, shmop, soap, sockets, sodium, sqlite3,
	standard, sysvmsg, sysvsem, sysvshm, tidy, tokenizer, xml, xmlreader, xmlrpc, xmlwriter, xsl, zip, zlib)
Setup: max_execution_time="10" memory_limit="512M" post_max_size="10500000" safe_mode="" session.save_handler="files"
	upload_max_filesize="10500000" 
Software: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips PHP/8.0.11 ()
	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.82 Safari/537.36
Server Software: MariaDB Server
	Server Version: 10.5.12-MariaDB
	Server SQL Mode: 
	Parameters: saglpi@127.0.0.1/glpi
	Host info: 127.0.0.1 via TCP/IP
	
 PHP version is at least 7.2.0 - Perfect!
 Sessions support is available - Perfect!
 Allocated memory > 64 Mio - Perfect!
 mysqli extension is installed
 ctype extension is installed
 fileinfo extension is installed
 json extension is installed
 mbstring extension is installed
 iconv extension is installed
 zlib extension is installed
 curl extension is installed
 gd extension is installed
 simplexml extension is installed
 intl extension is installed
 ldap extension is installed
 apcu extension is installed
 Zend OPcache extension is installed
 xmlrpc extension is installed
 exif extension is installed
 zip extension is installed
 bz2 extension is installed
 sodium extension is installed
 Database version seems correct (10.5.12) - Perfect!
 Timezones seems loaded in database
 The log file has been created successfully.
 Write access to /var/lib/glpi/files/_cache has been validated.
 Write access to /etc/glpi has been validated.
 Write access to /var/lib/glpi/files/_cron has been validated.
 Write access to /var/lib/glpi/files has been validated.
 Write access to /var/lib/glpi/files/_dumps has been validated.
 Write access to /var/lib/glpi/files/_graphs has been validated.
 Write access to /var/lib/glpi/files/_lock has been validated.
 Write access to /var/lib/glpi/files/_pictures has been validated.
 Write access to /var/lib/glpi/files/_plugins has been validated.
 Write access to /var/lib/glpi/files/_rss has been validated.
 Write access to /var/lib/glpi/files/_sessions has been validated.
 Write access to /var/lib/glpi/files/_tmp has been validated.
 Write access to /var/lib/glpi/files/_uploads has been validated.
 Write access to /var/lib/glpi/files/_marketplace has been validated.
 SELinux configuration is OK.

GLPI constants
 
GLPI_ROOT: /usr/share/glpi
GLPI_CONFIG_DIR: /etc/glpi
GLPI_HTMLAWED: /usr/share/glpi/vendor/htmlawed/htmlawed/htmLawed.php
GLPI_MARKETPLACE_DIR: /var/lib/glpi/files/_marketplace
GLPI_VAR_DIR: /var/lib/glpi/files
GLPI_LOG_DIR: /var/log/glpi
GLPI_FONT_FREESANS: /usr/share/fonts/gnu-free/FreeSans.ttf
GLPI_SYSTEM_CRON: 1
GLPI_INSTALL_MODE: RPM
GLPI_USE_CSRF_CHECK: 1
GLPI_CSRF_EXPIRES: 7200
GLPI_CSRF_MAX_TOKENS: 100
GLPI_USE_IDOR_CHECK: 1
GLPI_IDOR_EXPIRES: 7200
GLPI_ALLOW_IFRAME_IN_RICH_TEXT: 
GLPI_TELEMETRY_URI: https://telemetry.glpi-project.org
GLPI_NETWORK_MAIL: glpi@teclib.com
GLPI_NETWORK_SERVICES: https://services.glpi-network.com
GLPI_MARKETPLACE_PRERELEASES: 
GLPI_MARKETPLACE_ALLOW_OVERRIDE: 1
GLPI_MARKETPLACE_MANUAL_DOWNLOADS: 1
GLPI_USER_AGENT_EXTRA_COMMENTS: 
GLPI_AJAX_DASHBOARD: 1
GLPI_CALDAV_IMPORT_STATE: 0
GLPI_DEMO_MODE: 0
GLPI_FORCE_EMPTY_SQL_MODE: 1
GLPI_DOC_DIR: /var/lib/glpi/files
GLPI_CACHE_DIR: /var/lib/glpi/files/_cache
GLPI_CRON_DIR: /var/lib/glpi/files/_cron
GLPI_DUMP_DIR: /var/lib/glpi/files/_dumps
GLPI_GRAPH_DIR: /var/lib/glpi/files/_graphs
GLPI_LOCAL_I18N_DIR: /var/lib/glpi/files/_locales
GLPI_LOCK_DIR: /var/lib/glpi/files/_lock
GLPI_PICTURE_DIR: /var/lib/glpi/files/_pictures
GLPI_PLUGIN_DOC_DIR: /var/lib/glpi/files/_plugins
GLPI_RSS_DIR: /var/lib/glpi/files/_rss
GLPI_SESSION_DIR: /var/lib/glpi/files/_sessions
GLPI_TMP_DIR: /var/lib/glpi/files/_tmp
GLPI_UPLOAD_DIR: /var/lib/glpi/files/_uploads
GLPI_NETWORK_REGISTRATION_API_URL: https://services.glpi-network.com/api/registration/
GLPI_MARKETPLACE_PLUGINS_API_URI: https://services.glpi-network.com/api/glpi-plugins/
GLPI_I18N_DIR: /usr/share/glpi/locales
GLPI_VERSION: 9.5.6
GLPI_SCHEMA_VERSION: 9.5.6
GLPI_MIN_PHP: 7.2.0
GLPI_YEAR: 2021

Libraries
 
htmlawed/htmlawed version 1.2.5 in (/usr/share/glpi/vendor/htmlawed/htmlawed)
phpmailer/phpmailer version 6.1.6 in (/usr/share/glpi/vendor/phpmailer/phpmailer/src)
simplepie/simplepie version 1.5.6 in (/usr/share/glpi/vendor/simplepie/simplepie/library)
tecnickcom/tcpdf version 6.3.5 in (/usr/share/glpi/vendor/tecnickcom/tcpdf)
michelf/php-markdown in (/usr/share/glpi/vendor/michelf/php-markdown/Michelf)
true/punycode in (/usr/share/glpi/vendor/true/punycode/src)
iamcal/lib_autolink in (/usr/share/glpi/vendor/iamcal/lib_autolink)
sabre/dav in (/usr/share/glpi/vendor/sabre/dav/lib/DAV)
sabre/http in (/usr/share/glpi/vendor/sabre/http/lib)
sabre/uri in (/usr/share/glpi/vendor/sabre/uri/lib)
sabre/vobject in (/usr/share/glpi/vendor/sabre/vobject/lib)
laminas/laminas-cache in (/usr/share/glpi/vendor/laminas/laminas-cache/src)
laminas/laminas-i18n in (/usr/share/glpi/vendor/laminas/laminas-i18n/src)
laminas/laminas-serializer in (/usr/share/glpi/vendor/laminas/laminas-serializer/src)
monolog/monolog in (/usr/share/glpi/vendor/monolog/monolog/src/Monolog)
sebastian/diff in (/usr/share/glpi/vendor/sebastian/diff/src)
elvanto/litemoji in (/usr/share/glpi/vendor/elvanto/litemoji/src)
symfony/console in (/usr/share/glpi/vendor/symfony/console)
scssphp/scssphp in (/usr/share/glpi/vendor/scssphp/scssphp/src)
laminas/laminas-mail in (/usr/share/glpi/vendor/laminas/laminas-mail/src/Protocol)
laminas/laminas-mime in (/usr/share/glpi/vendor/laminas/laminas-mime/src)
rlanvin/php-rrule in (/usr/share/glpi/vendor/rlanvin/php-rrule/src)
blueimp/jquery-file-upload in (/usr/share/glpi/vendor/blueimp/jquery-file-upload/server/php)
ramsey/uuid in (/usr/share/glpi/vendor/ramsey/uuid/src)
psr/log in (/usr/share/glpi/vendor/psr/log/Psr/Log)
psr/simple-cache in (/usr/share/glpi/vendor/psr/simple-cache/src)
mexitek/phpcolors in (/usr/share/glpi/vendor/mexitek/phpcolors/src/Mexitek/PHPColors)
guzzlehttp/guzzle in (/usr/share/glpi/vendor/guzzlehttp/guzzle/src)
guzzlehttp/psr7 in (/usr/share/glpi/vendor/guzzlehttp/psr7/src)
wapmorgan/unified-archive in (/usr/share/glpi/vendor/wapmorgan/unified-archive/src)
paragonie/sodium_compat in (/usr/share/glpi/vendor/paragonie/sodium_compat/src)
phpCas version 1.4.0 in (/usr/share/pear)

SQL replicas
 
Not active

Notifications
 
Way of sending emails: SMTP+TLS (anonymous@10.10.1.72)

Mails receivers
 
Name: 'xxx@novreg.ru' Active: Yes
	Server: '{10.10.1.72/imap/ssl/novalidate-cert/norsh}Test' Login: 'xxx@gov.novreg.ru' Password: Yes
Name: 'xxxx@novreg.ru' Active: No
	Server: '{10.10.1.72/imap/ssl/novalidate-cert/norsh}' Login: 'xxxx@gov.novreg.ru' Password: Yes

Plugins list
 
	news                 Name: Alerts                         Version: 1.9.0      State: Enabled
	behaviors            Name: Behaviours                     Version: 2.5.0      State: Enabled
	dashboard            Name: Dashboard                      Version: 1.0.2      State: Enabled
	escalade             Name: Escalation                     Version: 2.6.2      State: Enabled
	formcreator          Name: Form Creator                   Version: 2.12.2     State: Enabled
	mreporting           Name: More Reporting                 Version: 1.7.3      State: Enabled
	moreticket           Name: More ticket                    Version: 1.6.1      State: Enabled
	mydashboard          Name: My Dashboard                   Version: 1.8.2      State: Enabled
	mycustomview         Name: MyCustomView                   Version: 1.2.1      State: Enabled
	ocsinventoryng       Name: OCS Inventory NG               Version: 1.7.3      State: Installed / not activated
	reports              Name: Reports                        Version: 1.14.1     State: Enabled
	screenshot           Name: Screenshot                     Version: 1.1.4      State: Enabled

[/code]
```

**Comment by BorisNovg** [comment]
Small attempt to debug the code (glpi 9.5.6 release version).
inc/mailcollector.class.php: line 1580->1581->1587->1592->1594->1587->1592->1594->1587->1592->1594...

... infinite loop

**Comment by BorisNovg** [comment]
Example:
[Message16335125161734170604.zip](https://github.com/glpi-project/glpi/files/7293469/Message16335125161734170604.zip)

**Comment by trasher** [comment]
Thanks for your feedback @BorisNovg.

Could you please try if the proposed fix in #9672 solves your issues? It may also fix #9600.

**Comment by BorisNovg** [comment]
Fix 89104ec (https://github.com/glpi-project/glpi/pull/9672) solves the problem.

I tested using the message [Message16335125161734170604.zip](https://github.com/glpi-project/glpi/files/7293469/Message16335125161734170604.zip).

In addition, I created a message with three different attachments, but having the same name without an extension.
[Message-test_3_different_attachments_with_the_same_filename_without_extension.zip](https://github.com/glpi-project/glpi/files/7303449/Message-test_3_different_attachments_with_the_same_filename_without_extension.zip)

![test_3_different_attachments_with_the_same_filename_without_extensions-1](https://user-images.githubusercontent.com/39726474/136390121-72c788dd-302d-46e4-b81a-a11b68a15714.gif)

![test_3_different_attachments_with_the_same_filename_without_extensions-2](https://user-images.githubusercontent.com/39726474/136391040-a90b50cd-fb4b-436a-95e0-5f640dfdbb07.gif)

All three files have been saved, their names are:
1234567890
1234567890_1
1234567890_2

Ok!

Also, I checked a letter in which two screenshots are attached inside the letter, different but with the same file name.
The file name is =?US-ASCII?B?0LjQt9C+0LHRgNCw0LbQtdC90LjQtS5wbmc=?=, which means изображение.png (screenshot.png in Russian).

![test_3_different_attachments_with_the_same_filename_without_extensions-3](https://user-images.githubusercontent.com/39726474/136392939-85f89c32-295e-42c8-8216-d56405e8977d.gif)

![test_3_different_attachments_with_the_same_filename_without_extensions-4](https://user-images.githubusercontent.com/39726474/136395183-488a5ddd-9baf-4230-b1ec-84b49e0c096d.gif)

As shown in the screenshot, 2 images have been successfully inserted into the message text.
Ok!

All five documents received by mail:
![test_3_different_attachments_with_the_same_filename_without_extensions-5](https://user-images.githubusercontent.com/39726474/136396421-3e6995cd-91e9-4224-ac7b-a2c1cfb72ddd.gif)

In addition, I sent copies of the last 538 messages from the my production system to the mail receiver. No errors were found.
